### PR TITLE
[ROCm][FEAT] Support AITER RMSNorm quantization fusion pass 

### PR DESCRIPTION
**Pull request by vllmellm**
## Purpose

This PR supports fusion pass for ROCM AITER by fusing `+rms_norm`, aiter rmsnorm ops, and `+quant_fp8`, vllm quantization custom ops.

**Benchmark Result**

| Metric                              | Without Fusion Pass | With Fusion Pass   |
|--------------------------------------|---------------------|-------------------|
| Successful requests                  | 500                 | 500               |
| Benchmark duration (s)               | 173.76              | 170.31            |
| Total input tokens                   | 520,558             | 520,558           |
| Total generated tokens               | 456,122             | 456,834           |
| Request throughput (req/s)           | 2.88                | 2.94              |
| Output token throughput (tok/s)      | 2,625.02            | 2,682.37          |
| Peak output token throughput (tok/s) | 7,924.00            | 7,410.00          |
| Peak concurrent requests             | 500.00              | 500.00            |
| Total token throughput (tok/s)       | 5,620.87            | 5,738.91          |
| Mean TTFT (ms)                       | 35,048.69           | 34,637.29         |
| Median TTFT (ms)                     | 28,413.54           | 28,841.97         |
| P99 TTFT (ms)                        | 91,625.66           | 90,845.96         |
| Mean TPOT (ms)                       | 167.64              | 170.28            |
| Median TPOT (ms)                     | 117.48              | 118.62            |
| P99 TPOT (ms)                        | 881.79              | 913.72            |
| Mean ITL (ms)                        | 114.17              | 113.99            |
| Median ITL (ms)                      | 57.83               | 61.15             |
| P99 ITL (ms)                         | 2,111.96            | 2,086.31          |

### benchmark setting
`
vllm bench serve \
--backend vllm \
--model "RedHatAI/Qwen3-14B-FP8-dynamic" \
--dataset-name random \
--num-prompts 500 \
--random-input-len 1000 \
--random-output-len 1000 \
--endpoint /v1/completions \
--random-range-ratio 0.9 \
`

## Test Plan

- Unit test has been added in `vllm/tests/compile/test_rocm_aiter_fusion.py` that verifies accuracy, replacement of the ops in cuda graph.
- end to end test using `RedHatAI/Qwen3-14B-FP8-dynamic` model

**environment setting**
Step 1: run vllm serve
`
vllm serve RedHatAI/Qwen3-14B-FP8-dynamic \
--compilation-config '{"pass_config": {"fuse_norm_quant": true, "fuse_act_quant": true, "eliminate_noops": true}, "custom_ops": ["+rms_norm", "+quant_fp8"]}' \
--port 9090 \
--trust-remote-code --swap-space 16 --distributed-executor-backend mp
`
Step 2: run lm_eval

`lm_eval --model local-completions
--tasks gsm8k \
--model_args model=RedHatAI/Qwen3-14B-FP8-dynamic,base_url=http://localhost:9090/v1/completions \
--trust_remote_code \
--num_fewshot 5 \
--batch_size 128`

### RedHatAI/Qwen3-14B-FP8-dynamic fusion pass (per token quantization fused with rmsnorm)
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.7612|±  |0.0117|
|     |       |strict-match    |     5|exact_match|↑  |0.8741|±  |0.0091|

### RedHatAI/Qwen3-14B-FP8-dynamic without fusion pass
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.7718|±  |0.0116|
|     |       |strict-match    |     5|exact_match|↑  |0.8741|±  |0.0091|

### Qwen/Qwen3-Coder-30B-A3B-Instruct-FP8 fusion pass (per group quantization fused with rmsnorm)

|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.8825|±  |0.0089|
|     |       |strict-match    |     5|exact_match|↑  |0.8931|±  |0.0085|

### Qwen/Qwen3-Coder-30B-A3B-Instruct-FP8 without fusion pass
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.8772|±  |0.0090|
|     |       |strict-match    |     5|exact_match|↑  |0.8908|±  |0.0086|

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

**Review by ProExpertProg — CHANGES_REQUESTED** [review]
Can we keep the same kinds of fusion contained to the same pass? RMSNormQuantFusionPass should contain all flavors of rms_norm + quant fusion

**Comment by vllmellm** [comment]
> Can we keep the same kinds of fusion contained to the same pass? RMSNormQuantFusionPass should contain all flavors of rms_norm + quant fusion

@ProExpertProg Do you mean to AVOID creating a separate `RMSNormAiterQuantFusionPass` ? and use the same `RMSNormQuantFusionPass`? if so, i'm afraid that would cause error for non ROCM platforms as the rmsnorms functions are that we registered on vllm library are accessed like`torch.ops.vllm.rocm_aiter_rms_norm.default` WOULD NOT be found in other platform and would cause errors and in order to avoid that we have to clutter the `RMSNORMQuantFusionPass` with if branches , in different functions like `RMSNormQuantFusionPass.uuid()` and `RMSNormQuantFusionPass.__init__()`,  
while having the RMSNormAiterQuantFusionPass separated we just need to have one single if statement in `pass_manager.py`. on the other hand, in future there will be other aiter patterns with different quantization functions to be merged this way it would be cleaner to keep the aiter fused passed separated?

File: tests/compile/test_fusion.py [hunk 1]
```diff
@@ -328,11 +250,216 @@ def test_fusion_rmsnorm_quant(
...
-            and model_class is not TestRmsnormGroupFp8QuantModel
-        ):
+        if not enable_rms_norm_custom_op:
             n_add_nodes = lambda g: sum(1 for _ in find_op_nodes(torch.ops.aten.add, g))
             # 7 = 1 (RMS) + 3x2 (3xRMS_ADD, 2 each)
             assert n_add_nodes(backend.graph_pre_pass) == 7
             assert n_add_nodes(backend.graph_post_pass) == 2
+
+
+class TestModelROCmAiter(torch.nn.Module):
```
Comment (ProExpertProg) [inline tests/compile/test_fusion.py]: Why can;t we just reuse the model and test definition above?

**Review by ProExpertProg — APPROVED** [review]

File: tests/compile/test_fusion.py (hunk 1 above)
Comment (vllmellm) [reply tests/compile/test_fusion.py]: Fixed, I reused the model but kept the test definition separate for simplicity. I also added a helper method to reduce redundancy in the test method.
